packages/lib/src/components/Card/components/CardInput/CardInput.test.tsx

```diff
@@ -78,19 +78,27 @@ describe('CardInput - Brands beneath Card Number field', () => {
         expect(wrapper.find('.adyen-checkout__card__brands__brand-wrapper--disabled')).toHaveLength(0);
     });
 
-    test('should disable the brands icons that are not detected', () => {
+    test('should hide all brand icons when brand is detected', () => {
         const detectedBrand = 'visa';
         const brandsIcons = [
             { name: 'visa', icon: 'visa.png' },
             { name: 'mc', icon: 'mc.png' },
             { name: 'amex', icon: 'amex.png' }
         ];
         const wrapper = mount(<CardInput i18n={i18n} brand={detectedBrand} brandsIcons={brandsIcons} />);
-        const brands = wrapper.find('.adyen-checkout__card__brands__brand-wrapper');
+        expect(wrapper.find('.adyen-checkout__card__brands--hidden')).toHaveLength(1);
+    });
 
-        expect(brands.at(0).is('.adyen-checkout__card__brands__brand-wrapper--disabled')).toBeFalsy();
-        expect(brands.at(1).is('.adyen-checkout__card__brands__brand-wrapper--disabled')).toBeTruthy();
-        expect(brands.at(2).is('.adyen-checkout__card__brands__brand-wrapper--disabled')).toBeTruthy();
+    test('should show all brand icons when no brand is detected', () => {
+        const detectedBrand = 'card';
+        const brandsIcons = [
+            { name: 'visa', icon: 'visa.png' },
+            { name: 'mc', icon: 'mc.png' },
+            { name: 'amex', icon: 'amex.png' }
+        ];
+        const wrapper = mount(<CardInput i18n={i18n} brand={detectedBrand} brandsIcons={brandsIcons} />);
+        const brandWrapper = wrapper.find('.checkout__card__brands');
+        expect(wrapper.find('.adyen-checkout__card__brands--hidden')).toHaveLength(0);
```

Comment:
Testing if the code has a classname does not mean that it works as expected - the CSS could be using the wrong properties. 
In this case you should check if the images were rendered in the DOM or not.

Reply:
I think they will still be in the DOM, since they're only animated away. I could check if their height is 0px, is that valid too?

Reply:
It was agreed to follow up on this in another PR